lib/internal/async_local_storage/async_context_frame.js

```diff
@@ -11,6 +12,19 @@ const {
 const AsyncContextFrame = require('internal/async_context_frame');
 const { AsyncResource } = require('async_hooks');
 
+class DisposableStore {
+  #oldFrame = undefined;
+
+  constructor(store, storage) {
+    this.#oldFrame = AsyncContextFrame.current();
```

Comment:
I guess by capturing/restoring only the owned store instead the complete frame the side effects to other ALS users could be avoided.

Reply:
Agreed, an operation on an AsyncLocalStorage should not affect other instances. For reference: https://github.com/tc39/proposal-async-context-disposable/issues/2